sklearn/naive_bayes.py

```diff
@@ -49,6 +51,12 @@ def _joint_log_likelihood(self, X):
         predict_proba and predict_log_proba.
         """
 
+    @abstractmethod
+    def _check_X(self, X):
```

Comment:
Having this abstract method added is breaking downstream code. 

Which is OK, but being the change hidden in this huge PR made it a bit more difficult than necessary to decipher/dig up if there is any relevant comments about the change, etc.

Reply:
Yes, we are under aware of impact on downstream projects. See #15992

I also think we neglect to consider people inheriting from our objects. Can you submit a PR to remove the abstractmethod for 0.22.1 perhaps?

Reply:
see #15996